src/Styleguide/Pages/Artist/Components/LoadingArea.tsx

```diff
@@ -0,0 +1,46 @@
+import Spinner from "Components/Spinner"
+import React from "react"
+import { State as PreloadLinkState } from "Router/PreloadLink"
+import styled from "styled-components"
+import { Subscribe } from "unstated"
+
+export const LoadingArea = props => {
+  return (
+    <Subscribe to={[PreloadLinkState]}>
+      {({ state: { isFetching } }: PreloadLinkState) => {
+        const loaderClass = isFetching ? "loading" : ""
+
+        return (
+          <React.Fragment>
+            <SpinnerToggle className={loaderClass} />
+            <Container className={loaderClass}>{props.children}</Container>
+          </React.Fragment>
+        )
+      }}
+    </Subscribe>
```

Comment:
For example here. Import `PreloadLinkState` and subscribe, and now we've connected our main content area to the current state of the route request which enables some nice controlled behavior.

Reply:
Doesn’t that tie this component to its parent? Couldn’t this be done as well by taking a `isLoading` prop and having the parent pass in its `isFetching` state?

Reply:
The issue is that we're working across distant areas in the tree which make this sort of passing impractical and rigid. Will elaborate in Slack.